internal/component/pyroscope/java/loop.go

```diff
@@ -63,6 +71,16 @@ func newProfilingLoop(pid int, target discovery.Target, logger log.Logger, profi
 		return p
 	}
 
+	p.wg.Add(1)
+	go func() {
+		defer p.wg.Done()
+		// Clean-up files that weren't removed by a previous instance of alloy
+		err := p.cleanupOldJFRFiles(jfrFileName)
```

Comment:
Is it possible that those files were created by something other than Alloy? And that deleting them could interfere with this other process?